src/main/java/org/opensearch/ad/model/AnomalyResultBucket.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.ad.model;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.search.aggregations.bucket.composite.CompositeAggregation.Bucket;
+import org.opensearch.search.aggregations.metrics.InternalMax;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Represents a single bucket when retrieving top anomaly results for HC detectors
+ */
+public class AnomalyResultBucket implements ToXContentObject, Writeable {
+    public static final String BUCKETS_FIELD = "buckets";
+    public static final String KEY_FIELD = "key";
+    public static final String DOC_COUNT_FIELD = "doc_count";
+    public static final String MAX_ANOMALY_GRADE_FIELD = "max_anomaly_grade";
+
+    private final Map<String, Object> key;
+    private final int docCount;
+    private final double maxAnomalyGrade;
+
+    public AnomalyResultBucket(
+            Map<String, Object> key,
```

Comment:
`Map<String, Object>` is the class for `key` in class `org.opensearch.search.aggregations.bucket.composite.CompositeAggregation.Bucket` (called in line 57 here). Prefer to keep this way instead of hard casting to a string.